tests/compute/test_heterograph.py

```diff
@@ -2503,6 +2503,82 @@ def test_frame_device(idtype):
     assert F.context(ng._node_frames[0]._columns['hh'].storage) == F.ctx()
     assert F.context(ng._edge_frames[0]._columns['h'].storage) == F.cpu()
 
+@parametrize_dtype
+def test_create_block(idtype):
+    block = dgl.create_block(([0, 1, 2], [1, 2, 3]), idtype=idtype, device=F.ctx())
+    assert block.num_src_nodes() == 3
+    assert block.num_dst_nodes() == 4
+    assert block.num_edges() == 3
+
+    block = dgl.create_block(([0, 1, 2], [1, 2, 3]), (4, 5), idtype=idtype, device=F.ctx())
+    assert block.num_src_nodes() == 4
+    assert block.num_dst_nodes() == 5
+    assert block.num_edges() == 3
+
+    sx = F.randn((4, 5))
+    dx = F.randn((5, 6))
+    ex = F.randn((3, 4))
+    block.srcdata['x'] = sx
+    block.dstdata['x'] = dx
+    block.edata['x'] = ex
+
+    g = dgl.block_to_graph(block)
+    assert g.num_src_nodes() == 4
+    assert g.num_dst_nodes() == 5
+    assert g.num_edges() == 3
+    assert g.srcdata['x'] is sx
+    assert g.dstdata['x'] is dx
+    assert g.edata['x'] is ex
+
+    block = dgl.create_block({
+        ('A', 'AB', 'B'): ([1, 2, 3], [2, 1, 0]),
+        ('B', 'BA', 'A'): ([2, 3], [3, 4])},
+        idtype=idtype, device=F.ctx())
+    assert block.num_src_nodes('A') == 4
+    assert block.num_src_nodes('B') == 4
+    assert block.num_dst_nodes('B') == 3
+    assert block.num_dst_nodes('A') == 5
+    assert block.num_edges('AB') == 3
+    assert block.num_edges('BA') == 2
+
+    block = dgl.create_block({
+        ('A', 'AB', 'B'): ([1, 2, 3], [2, 1, 0]),
+        ('B', 'BA', 'A'): ([2, 3], [3, 4])},
+        num_nodes=({'A': 5, 'B': 5}, {'A': 6, 'B': 4}),
+        idtype=idtype, device=F.ctx())
```

Comment:
Test for number of nodes only partially specified? E.g. only specified for one of the node types, only specified for one of the input/output nodes.

Reply:
Just noticed that our `dgl.heterograph()` did not handle partially-specified `num_nodes_dict` as well:
```
In [2]: g = dgl.heterograph({('A', 'AB', 'B'): ([0,1,2],[1,2,3])}, num_nodes_dict={'A': 4})
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-3b6e4a670ddf> in <module>
----> 1 g = dgl.heterograph({('A', 'AB', 'B'): ([0,1,2],[1,2,3])}, num_nodes_dict={'A': 4})

~/miniconda3/lib/python3.7/site-packages/dgl-0.6-py3.7-linux-x86_64.egg/dgl/convert.py in heterograph(data_dict, num_nodes_dict, idtype, device)
    322                                ' the max ID in the data, but got {} and {}.'.format(
    323                                    sty, num_nodes_dict[sty], urange - 1))
--> 324             if num_nodes_dict[dty] < vrange:
    325                 raise DGLError('The given number of nodes of node type {} must be larger than'
    326                                ' the max ID in the data, but got {} and {}.'.format(

KeyError: 'B'
```